join-domain/elx/sssd/files/join.sh

```diff
@@ -0,0 +1,111 @@
+#!/bin/bash
+set -eu
+#
+# Script to join host to domain
+#
+#################################################################
+JOIN_DOMAIN="${JOIN_DOMAIN:-UNDEF}"
+JOIN_OU="${JOIN_OU:-}"
+JOIN_USER="${JOIN_USER:-Administrator}"
+JOIN_CNAME="${JOIN_CNAME:-UNDEF}"
+PWCRYPT="${ENCRYPT_PASS:-UNDEF}"
+PWUNLOCK="${ENCRYPT_KEY:-UNDEF}"
+CLIENT_OSNAME="$(
+  awk -F "=" '/^NAME/{ print $2}' /etc/os-release |
+  sed 's/"//g'
+)"
+CLIENT_OSVERS="$(
+  awk -F "=" '/^VERSION_ID/{ print $2 }' /etc/os-release |
+  sed 's/"//g'
+)"
+
+# Get clear-text password from crypt
+function PWdecrypt {
+  local PWCLEAR
+
+  # Get cleartext password-string
+  PWCLEAR=$(
+    echo "${PWCRYPT}" | \
+    openssl enc -aes-256-cbc -md sha256 -a -d -salt -pass pass:"${PWUNLOCK}"
+  )
+
+  # This sucks, but...
+  # shellcheck disable=SC2181
+  if [[ $? -ne 0 ]]
+  then
+    echo ""
+    return 1
+  else
+    echo "${PWCLEAR}"
+    return 0
+  fi
+}
+
+# Make sure domain is discoverable
+function IsDiscoverable {
+  if [[ $( realm discover "${JOIN_DOMAIN}" > /dev/null 2>&1 )$? -eq 0 ]]
+  then
+    printf "The %s domain is discoverable\n" "${JOIN_DOMAIN}"
+    return 0
+  else
+    printf "The %s domain is not discoverable. Aborting...\n" "${JOIN_DOMAIN}"
+    return 1
+  fi
+}
+
+# Try to join host to domain
+function JoinDomain {
+
+  # Toggle SELinux if necessary
+  if [[ $( getenforce ) == "Enforcing" ]]
+  then
+    SEL_TARG="1"
+    printf "Toggling SELinux mode... "
+    setenforce 0 || (echo "FAILED" ; exit 1 )
+    echo SUCCESS
+  else
+    SEL_TARG=0
+  fi
+
+  if [[ -z ${JOIN_OU} ]]
+  then
+    printf "Joining to %s... " "${JOIN_DOMAIN}"
+    # shellcheck disable=SC2005
+    echo "$( PWdecrypt )" | \
```

Comment:
The [shellcheck doc](https://www.shellcheck.net/wiki/SC2005) suggests we should be able to just call the function, since the function echoes the value we don't need this wrapping echo... If that works, same change on line 85-86

```suggestion
    PWdecrypt | \
```

Reply:
Yeah,  shellcheck bitched at me about it. However, in testing (on EL7), doing the naked `PWdecrypt` resulted in failure (thus the commit-note in commit acc000362b5f614897719cc31202afb9cad44915 ☹️). Unfortunately, unlike with PBIS, there is no flag for `realm join …` that allows you to just include the string as an argument in the command-invocation, itself – you have to resort to an `echo <STRING> | realm join …` kludgery. It would seem that shellcheck is wrong about the method: whether that's a function of the elderliness of  EL7's version of BASH or how `realm join` reads from `STDIN` I have no clue, but it was the necessary method.

Reply:
That's fine, just wanted to check we at least tried.